pkg/cri/containerd.go

```diff
@@ -254,22 +324,17 @@ func (c *Containerd) createContainer(image string,
 
 		// Check if AWS CLI is installed
 		var cmd *exec.Cmd
-		var awsPath string
-
-		if awsPath, err = exec.LookPath("aws"); err == nil {
-			// Get ECR password
-			cmd = exec.Command(awsPath, "ecr", "get-login-password", "--region", "us-east-2")
-			ecrPasswordBytes, err := cmd.Output()
-			if err != nil {
-				// Return an error if neither file exists
-				journal.Error("Failed to pull image: Error retrieving ECR password",
-					"containerName", containerName,
-					"image", image)
-				return nil, err
-			}
-			ecrPassword := strings.TrimSpace(string(ecrPasswordBytes))
-			cmd = exec.Command(ctrPath, "-n", "k8s.io", "images", "pull", "--user", fmt.Sprintf("AWS:%s", ecrPassword), image)
-		} else {
+		var ecr_region string
+		// First check if it's an ECR image
+		ecr_region, err = extractECRRegion(image)
+		if err != nil { journal.Debug("extractECRRegion return error, %v", err) }
+		
+		if err == nil {
+			// If it is an ECR image, get the authenticated pull command
+			cmd, err = getECRPullCommand(image, ctrPath, ecr_region)
+			if err != nil { journal.Debug("ECR region is %s, getECRPullCommand return error, %v",ecr_region, err) }
+		} 
+		if err != nil {
```

Comment:
```suggestion
                 var ecrRegion string
		// First check if it's an ECR image
		ecrRegion, err = extractECRRegion(image)
		if err != nil { journal.Debug("extractECRRegion return error, %v", err) }
		else {
		        // If it is an ECR image, get the authenticated pull command
			cmd, err = getECRPullCommand(image, ctrPath, ecrRegion)
			if err != nil { journal.Debug("ECR region is %s, getECRPullCommand return error, %v", ecrRegion, err) }
		}
		if err != nil {
```

mainly use camelCase and not snake_case